Capri.Web/src/components/myProporsals.vue

```diff
@@ -0,0 +1,298 @@
+                    description:
+                        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce nulla lacus, facilisis vel felis id, porta blandit nulla. Quisque congue congue orci elementum tristique. Integer consequat est a nibh mollis aliquam. Integer leo justo, posuere fermentum euismod sit amet, blandit sagittis elit. Suspendisse laoreet massa nec neque tincidunt, in ullamcorper urna sollicitudin. Mauris vitae arcu bibendum, bibendum nisi vitae, aliquet nunc. In tincidunt purus sed lacus tincidunt facilisis. ',
+                    thesisType: 'Bachelor',
+                    studyType: 'Part time',
+                    freeSlots: 2,
+                    students: ['Jan 2', 'Jan 23'],
+                    maxStudents: 4,
+                },
+                {
+                    title: 'CaPri System',
+                    promoter: 'Jerzy Nawrocki',
+                    description:
+                        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce nulla lacus, facilisis vel felis id, porta blandit nulla. Quisque congue congue orci elementum tristique. Integer consequat est a nibh mollis aliquam. Integer leo justo, posuere fermentum euismod sit amet, blandit sagittis elit. Suspendisse laoreet massa nec neque tincidunt, in ullamcorper urna sollicitudin. Mauris vitae arcu bibendum, bibendum nisi vitae, aliquet nunc. In tincidunt purus sed lacus tincidunt facilisis. ',
+                    thesisType: 'Master',
+                    studyType: 'Full time',
+                    freeSlots: 3,
+                },
+                {
+                    title: 'CaPri System',
+                    promoter: 'Jerzy Nawrocki',
+                    description:
+                        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce nulla lacus, facilisis vel felis id, porta blandit nulla. Quisque congue congue orci elementum tristique. Integer consequat est a nibh mollis aliquam. Integer leo justo, posuere fermentum euismod sit amet, blandit sagittis elit. Suspendisse laoreet massa nec neque tincidunt, in ullamcorper urna sollicitudin. Mauris vitae arcu bibendum, bibendum nisi vitae, aliquet nunc. In tincidunt purus sed lacus tincidunt facilisis. ',
+                    thesisType: 'Master',
+                    studyType: 'Full time',
+                    freeSlots: 3,
+                },
+            ],
+        };
+    }
+    public handleClick(value) {
+        this.popup.title = value.title;
+        this.popup.promoter = value.promoter;
+        this.popup.studyType = value.studyType;
+        this.popup.thesisType = value.thesisType;
+        this.popup.freeSlots = value.freeSlots;
+        this.popup.description = value.description;
+        this.popup.students = value.students;
+        this.dialog = true;
+    }
+    public addStudent() {
+        if (this.popup.students === undefined) {
+            this.popup.students = [];
+        }
+        this.popup.students.push('');
```

Comment:
I think better way of using that will be:
`if(!this.popup.students) {
this.popup.students = [];
}
this.popup.students.push('');`